python/src/boombox/boombox.py

```diff
@@ -0,0 +1,472 @@
+                data_type = "s"
+            case "u":
+                data_type = "u"
+            case "f":
+                data_type = "f"
+            case other:
+                warnings.warn(
+                    f"Arrays of dtype.kind == {other} not allowed, supported kinds are 'i', 'u', 'f', casting to 'f'."
+                )
+                data_type = "f"
+
+        match dtype.itemsize:
+            case 1 | 2 | 4 | 8:
+                bit_size = str(dtype.itemsize * 8)
+            case other:
+                warnings.warn(
+                    f"Item size {dtype.itemsize} not allowed, supported item sizes are 1, 2, 4 and 8, casting to 4"
+                )
+                bit_size = "32"
+
+        match dtype.byteorder:
+            case "<":
+                endian = "le"
+            case ">":
+                endian = "be"
+            case "=":
+                endian = "le" if sys.byteorder == "little" else "be"
+            case "|":
+                endian = ""
+
+        sample_format = data_type + bit_size + endian
+        # trick to ensure that sample_format is of AudioSampleFormat type
+        audio_sample_formats: tuple[AudioSampleFormat, ...] = get_args(
+            AudioSampleFormat
+        )
+        assert sample_format in audio_sample_formats
+
+        new_dtype = self._sample_format_to_dtype(sample_format)
+
+        return sample_format, new_dtype
+
+    def _sample_format_to_dtype(self, sample_format: AudioSampleFormat) -> np.dtype:
+        type_mapping = {"s": "i", "u": "u", "f": "f"}
+
+        if len(sample_format) == 2:  # Handle 8-bit case (without endian)
+            data_type = sample_format[0]
+            dtype_str = type_mapping[data_type] + "1"
+        else:
+            data_type = sample_format[0]
+            bit_size = sample_format[1:-2]
+            endian = sample_format[-2:]
+
+            # numpy doesn't support 24-bit size values
+            bit_size = 32 if bit_size == "24" else int(bit_size)
+            endian_symbol = "<" if endian == "le" else ">"
+            dtype_str = (
+                endian_symbol + type_mapping[data_type] + str(int(bit_size) // 8)
+            )
+
+        return np.dtype(dtype_str)
+
+    def _audio_bytes_to_array(
+        self, data: bytes, sample_format: AudioSampleFormat
+    ) -> np.ndarray:
+        dtype = self._sample_format_to_dtype(sample_format)
+        if sample_format not in ["s24le", "u24le", "s24be", "u24be"]:
+            return np.frombuffer(data, dtype)
+        else:
+            # numpy doesn't support 24-bit samples, this transforms them to 32-bit.
+            endian = "little" if sample_format[-2:] == "le" else "big"
+            is_signed = sample_format[0] == "s"
+            return np.array(
+                int.from_bytes(data[i : i + 3], endian, signed=is_signed)
+                for i in range(0, len(data), 3)
+            )
+
+    def _deserialize_packet(self, packet: dict[Atom, Any]) -> AudioPacket | VideoPacket:
```

Comment:
I believe some of the functions (this one, but also `_sample_format_to_dtype`, `_audio_bytes_to_array` and perhaps some others as well) could be `@staticmethod`s